tests/test_smoke.py

```diff
@@ -383,7 +383,7 @@ def test_aws_region():
             f'sky exec {name} \'echo $SKYPILOT_CLUSTER_INFO | jq .region | grep us-east-2\'',
             f'sky logs {name} 2 --status',  # Ensure the job succeeded.
             # A user program should not access SkyPilot runtime env python by default.
-            f'sky exec {name} \'which python | grep {constants.SKY_REMOTE_PYTHON_ENV_NAME} || exit 1\'',
+            f'sky exec {name} \'which python | grep {constants.SKY_REMOTE_PYTHON_ENV_NAME} && exit 1\'',
```

Comment:
hmm, actually, when `which python | grep {constants.SKY_REMOTE_PYTHON_ENV_NAME}` is False, this entire command will still have return code `1`, and supposedly this line for the test should not work either. 

Just tested: `sky launch -c test --cloud aws "which python | grep skypilot-runtime && exit 1" --cpus 2 --memory 2+` and indeed it has return code 1.

The following should work?
```suggestion
            f'sky exec {name} \'which python | grep {constants.SKY_REMOTE_PYTHON_ENV_NAME} && exit 1 || true \'',
```

Reply:
Good catch